src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs

```diff
@@ -366,6 +363,24 @@ public override object Clone()
             this.logger.LogTrace("(-)");
         }
 
+        /// <summary>
+        /// Whether peer should receive only blocks.
+        /// </summary>
+        /// <param name="peer">Peer to check.</param>
+        /// <returns>Whether peer should receive only blocks.</returns>
+        private bool IsBlocksOnly(INetworkPeer peer)
+        {
+            this.logger.LogTrace("({0}:'{1}'", nameof(peer), peer.RemoteSocketEndpoint);
+            bool isBlocksOnly = !this.connectionManager.ConnectionSettings.RelayTxes;
+
+            // Allow whitelisted peers to send data other than blocks in blocks only mode if whitelistrelay is true
+            if (peer.Behavior<ConnectionManagerBehavior>().Whitelisted && this.mempoolManager.mempoolSettings.WhiteListRelay)
```

Comment:
This can be improved, when calling the method AttachCore you could find this behaviour and set a flag on the current behaviour, then we wont need to look it up every time a trx is incoming.